sparse_strips/vello_common/src/glyph.rs

```diff
@@ -0,0 +1,235 @@
+        self.run.normalized_coords = bytemuck::cast_slice(coords);
+        self
+    }
+
+    /// Consumes the builder and fills the glyphs with the current configuration.
+    pub fn fill_glyphs(self, glyphs: impl Iterator<Item = Glyph>) {
+        self.render(glyphs, Style::Fill);
+    }
+
+    /// Consumes the builder and strokes the glyphs with the current configuration.
+    pub fn stroke_glyphs(self, glyphs: impl Iterator<Item = Glyph>) {
+        self.render(glyphs, Style::Stroke);
+    }
+
+    fn render(self, glyphs: impl Iterator<Item = Glyph>, style: Style) {
+        let font =
+            skrifa::FontRef::from_index(self.run.font.data.as_ref(), self.run.font.index).unwrap();
+        let outlines = font.outline_glyphs();
+        let size = Size::new(self.run.font_size);
+        let hinting_instance = if self.run.hint {
+            // Only transformations including uniform scaling and translation can be hinted.
+            // Rotated, skewed, and other transformations cannot be hinted.
+            let [a, b, c, d, _, _] = self.run.transform.as_coeffs();
+            if a == d && b == 0.0 && c == 0.0 {
+                // TODO: Cache hinting instance.
+                HintingInstance::new(&outlines, size, self.run.normalized_coords, HINTING_OPTIONS)
+                    .ok()
+            } else {
+                None
+            }
+        } else {
+            None
+        };
+
+        let render_glyph = match style {
+            Style::Fill => GlyphRenderer::fill_glyph,
+            Style::Stroke => GlyphRenderer::stroke_glyph,
+        };
+        // Re-use the same `path` allocation for each glyph.
+        let mut path = OutlinePath(BezPath::new());
+        for glyph in glyphs {
+            path.0.truncate(0);
+            let draw_settings = if let Some(hinting_instance) = &hinting_instance {
+                DrawSettings::hinted(hinting_instance, false)
+            } else {
+                DrawSettings::unhinted(size, self.run.normalized_coords)
+            };
+            let Some(outline) = outlines.get(GlyphId::new(glyph.id)) else {
+                continue;
+            };
+            if outline.draw(draw_settings, &mut path).is_err() {
+                continue;
+            }
+            let mut transform = Affine::translate(Vec2::new(glyph.x as f64, glyph.y as f64));
+            if let Some(glyph_transform) = self.run.glyph_transform {
+                transform *= glyph_transform;
+            }
+
+            render_glyph(
+                self.renderer,
+                PreparedGlyph::Outline(OutlineGlyph {
+                    path: &path.0,
+                    local_transform: transform,
+                }),
+            );
+        }
+    }
+}
+
+enum Style {
+    Fill,
+    Stroke,
+}
+
+/// A sequence of glyphs with shared rendering properties.
+#[derive(Clone, Debug)]
+struct GlyphRun<'a> {
+    /// Font for all glyphs in the run.
+    pub font: Font,
```

Comment:
It's not immediately obvious to me why the fields of this private struct are public. But OTOH, if clippy isn't complaining about it, that's probably my lack of understanding here.

Reply:
Nice catch! Made private in https://github.com/linebender/vello/pull/883/commits/6e661a49209423ebbbeb76ea79c1f97204380907